### PR TITLE
Implement store to reduce the redundancies

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
- Implement Vuex
- Implement dynamic routes
- Make a common component for the Sports 
- Remove the redundant components
- Remove the individual components
- Rename the services
- Remove unnecessary routes

### CODE REVIEW

File: web-app/src/components/carrom-board/CarromBoardScoreModal.vue [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+      return winnerTeam && team.id === winnerTeam.id;
+    }
+
+    public isRoundWinner(activity: ActivityInterface, team: string) {
+      if (team.toLowerCase() === 'home' && activity.team.name === this.fixture.homeTeam.name) {
+        return true;
+      }
+
+      if (team.toLowerCase() === 'away' && activity.team.name === this.fixture.awayTeam.name) {
```
Comment (SAGARACH65) [inline web-app/src/components/carrom-board/CarromBoardScoreModal.vue]:
```suggestion
     return team.toLowerCase() === 'away' && activity.team.name === this.fixture.awayTeam.name
```

File: web-app/src/components/futsal/FutsalScoreModal.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    get isFixturePlayed() {
+      return isFixturePlayed(this.fixture);
+    }
+
+    get isFixtureCancelled() {
+      return isFixtureCancelled(this.fixture);
+    }
+
+    get fixtureDate() {
+      return this.fixture && dateFns.format(new Date(this.fixture.date), 'dddd MMM D, YYYY, hh:mm A');
```
Comment (SAGARACH65) [inline web-app/src/components/futsal/FutsalScoreModal.vue]: Should we add the date formats in a constant file?

File: web-app/src/services/tournaments.ts [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        id: tournament.id,
+        name: tournament.name,
+        shortName: tournament.shortName
+      };
+    });
+
+    seasons = seasons.concat(tournament.seasons);
+  });
+
+  seasons.sort((a, b) => differenceInDays(new Date(a.startDate), new Date(b.startDate))).reverse();
```
Comment (SAGARACH65) [inline web-app/src/services/tournaments.ts]:
```suggestion
return  seasons.sort((a, b) => differenceInDays(new Date(a.startDate), new Date(b.startDate))).reverse();
```

File: web-app/src/services/tournaments.ts [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+ * @export
+ * @param {FixtureInterface[]} fixturesList
+ * @param {number} [limit=0]
+ * @returns {FixtureInterface[]}
+ */
+function getFixtures(fixturesList: FixtureInterface[], limit: number = 0): FixtureInterface[] {
+  let fixtures = fixturesList
+    .filter(fixture => ['played', 'forfeited'].indexOf(fixture.status.toLowerCase()) < 0)
+    .sort((a, b) => {
+      return compareAsc(a.date, b.date);
```
Comment (SAGARACH65) [inline web-app/src/services/tournaments.ts]:
```suggestion
       compareAsc(a.date, b.date);
```
Let's remove the curly braces and the return statement as well

File: web-app/src/services/tournaments.ts [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+ *
+ * @param {FixtureInterface[]} fixturesList
+ * @param {number} [limit=0]
+ * @returns {FixtureInterface[]}
+ */
+function getResults(fixturesList: FixtureInterface[], limit: number = 0): FixtureInterface[] {
+  let results = fixturesList
+    .filter(fixture => ['played', 'forfeited'].indexOf(fixture.status.toLowerCase()) >= 0)
+    .sort((a, b) => {
+      return compareDesc(a.date, b.date);
```
Comment (SAGARACH65) [inline web-app/src/services/tournaments.ts]: .

**Comment by sanjeevkpandit** [comment]
These changes are from the old implementation. This will change gradually once the API is built.

**Pull request merged**